Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Miden-node Dockerfile
+
+#### Setup image builder
+FROM rust:1.75.0-bullseye AS builder
+
+# Install dependencies
+RUN apt-get update && apt-get -y upgrade
+RUN apt-get install -y gcc llvm clang bindgen pkg-config
+
+# Setup workdir
+WORKDIR /app
+COPY . miden-node
+RUN cd miden-node && make
+
+### Run Miden-Node
+FROM ubuntu:22.04
+RUN apt-get update && apt-get -y upgrade && apt-get install -y make libssl-dev libsqlite3-dev curl
+COPY --from=builder /app/miden-node/scripts/start-miden-node.sh start-miden-node.sh
+COPY --from=builder /app/miden-node/node/miden-node.toml miden-node.toml
+COPY --from=builder /app/miden-node/node/genesis.toml genesis.toml
+COPY --from=builder /usr/local/cargo/bin/miden-node /usr/local/bin/miden-node
+RUN chmod +x start-miden-node.sh
+EXPOSE 57291
+CMD [ "./start-miden-node.sh" ]
```

Comment:
This deserves some documentation. Specifically why this is sufficient as a PID 1

Reply:
Could you please clarify what should be documented, I believe that the node should be the running process.

Reply:
The PID 1 is special, it has to handle zombie processes and signals properly, otherwise the container misbehaves. This is okay as long as we don't spawn new processes on our architecture, and this should be documented. Some like:

```
# Miden node does not spawn sub-processes, so it can be used as the PID1
```

Another approach is to just fix the node to actually do the zombie cleanup

Reply:
What do you mean by "fixing the node to do the zombie cleanup" ?

Reply:
Here are some docs:

https://github.com/docker-library/official-images?tab=readme-ov-file#init
https://github.com/krallin/tini/issues/8
https://www.padok.fr/en/blog/docker-processes-container